docs/src/creating-packages.md

```diff
@@ -376,6 +369,34 @@ In fact, this form of type piracy is one of the most standard use cases for exte
     know about them, the extensions will not load. This is because the manifest lacks some information that tells Julia
     when it should load these packages. So make sure you use a manifest generated at least the Julia version you are using.
 
+### Behavior of extensions
+
+A user that depends only on `Plotting` will not pay the cost of the "extension" inside the `PlottingContourExt` module.
+It is only when the `Contour` package actually gets loaded that the `PlottingContourExt` extension is loaded
+and provides the new functionality.
+
+In our example, the new functionality is an additional _method_ for an existing _function_ from the parent package.
+Implementing such methods is among the most standard use cases of package extensions.
+Within the main package, the function to extend can even be defined with zero methods, as follows:
+
+```julia
+function plot end
+```
+
+!!! note
+    If one considers `PlottingContourExt` as a completely separate package, it could be argued that defining `Plotting.plot(c::Contour.ContourCollection)` is
+    [type piracy](https://docs.julialang.org/en/v1/manual/style-guide/#Avoid-type-piracy) since `PlottingContourExt` _owns_ neither the function `Plotting.plot` nor the type `Contour.ContourCollection`.
+    However, for extensions, it is ok to assume that the extension owns the functions in its parent package.
+
+In other situations, one may need to define new symbols in the extension (types, structs, functions, etc.) instead of reusing those from the parent.
+Such symbols are created in a separate module corresponding to the extension (`PlottingContourExt`), and not in the parent itself (`Plotting`).
+To retrieve this module and the symbols it contains for outside use, all it takes is a call to [`Base.get_extension`](@ref).
+Here is an example with a custom type defined in the extension:
+
+```julia
+const ContourPlotType = Base.get_extension(Plotting, :PlottingContourExt).ContourPlotType
+```
```

Comment:
Two things make me uneasy in this code example:
1. it does not account for the possibility of `get_extension` returning `nothing` if the extension is not loaded, and
2. I'm not sure whether the most common use case would be for a package to get access to its own extension, or for client code to access the extension of a package it depends on.

If the former is common enough, it might be useful to showcase an example along the lines of:
```julia
ext = Base.get_extension(@__MODULE__, :PlottingContourExt)
if !isnothing(ext)
   ContourPlotType = ext.ContourPlotType
end
```

As it stands, the example provided here hints at the latter use case, where code in a third-party package tries to access the extension to another package. I'm not sure this would be recommended practice:
- how can the developer of the 3rd party package know what is actually "exposed" by the extension as its "API"? In other words, is it safe to access `ContourPlotType` from the extension, or will its name change in the next match release?
- in any case, this "leaks" the extension name, which needs to be known for the 3rd party package to access it. I think I have seen somewhere that extension names should generally be considered internal implementation details and not be part of an API of the main package.

All in all, I'd be more comfortable if we advertised more prominently a use case in which the main package tries to gain access to its own extension.